dev/CommonStyles/TestUI/CompactPage.xaml.cs

```diff
@@ -0,0 +1,104 @@
+        {
+            if (item != null)
+            {
+                // Avoid Height because of CornerRadius
+                if (Math.Abs(item.ActualHeight - expectHeight) > 0.5)
+                {
+                    simpleVerify.IsEqual(item.ActualHeight.ToString(), expectHeight, controlName + ".ActualHeight=" + expectHeight);
+                }
+            }
+            else
+            {
+                simpleVerify.IsTrue(false, "Can't find " + controlName);
+            }
+        }
+
+        private void VerifyChildHeight(SimpleVerify simpleVerify, FrameworkElement item, string childXName, int expectHeight, string controlName)
+        {
+            FrameworkElement child = GetChild(item, childXName);
+            VerifyHeight(simpleVerify, child, expectHeight, controlName + "." + childXName);
+        }
+
+        private FrameworkElement GetChild(FrameworkElement item, string childXName)
+        {
+            FrameworkElement root = (item == null) ? null: (FrameworkElement)VisualTreeHelper.GetChild(item, 0);
+            FrameworkElement child = (root == null) ? null: (FrameworkElement)root.FindName(childXName);
+            return child;
+        }
+        private void RunTest_Click(object sender, RoutedEventArgs e)
+        {           
+            SimpleVerify simpleVerify = new SimpleVerify();
+            FrameworkElement item = ListView.FindElementOfTypeInSubtree<ListViewItem>();
+            VerifyHeight(simpleVerify, item, 32, "ListViewItem");
+            item = TreeView.FindElementOfTypeInSubtree<TreeViewItem>();
+            VerifyHeight(simpleVerify, item, 24, "TreeViewItem");
+
+            VerifyHeight(simpleVerify, NavItem1, 32, "NavigationViewItem");
+
+            VerifyChildHeight(simpleVerify, TextBox, "BorderElement", 24, "TextBox");
+            VerifyChildHeight(simpleVerify, PasswordBox, "BorderElement", 24, "PasswordBox");
+
+            // AutoSuggestBox used TextBox directly, so we check TextBox
+            FrameworkElement child = GetChild(AutoSuggestBox, "TextBox");
+            VerifyChildHeight(simpleVerify, child, "BorderElement", 24, "AutoSuggestBox.TextBox");
+
+            VerifyChildHeight(simpleVerify, RichEditBox, "BorderElement", 24, "RichEditBox");
+            VerifyChildHeight(simpleVerify, ComboBox, "Background", 24, "TextBox");
+
+            VerifyChildHeight(simpleVerify, TimePicker, "FirstPickerHost", 24, "TimePicker");
+            VerifyChildHeight(simpleVerify, TimePicker, "SecondPickerHost", 24, "TimePicker");
+            VerifyChildHeight(simpleVerify, TimePicker, "ThirdPickerHost", 24, "TimePicker");
+
+            VerifyChildHeight(simpleVerify, DatePicker, "DayTextBlock", 24, "DatePicker");
+            VerifyChildHeight(simpleVerify, DatePicker, "MonthTextBlock", 24, "DatePicker");
+            VerifyChildHeight(simpleVerify, DatePicker, "YearTextBlock", 24, "DatePicker");
+
```

Comment:
Missing validation of text size?

Reply:
Everything is ControlContentThemeFontSize, and we don't extra testing for it.
If the font size is wrong, the Height should be wrong and should be caught by the test case.